fastify/routes/unsafeFileUpload.js

```diff
@@ -20,10 +21,10 @@ module.exports = async function route(fastify, options) {
       const inputs = utils.getInput(req, key, params);
       // We sometimes use these routes just to test UFU and there is no input
       // Just return 'done' in that case
-      if (inputs) {
-        return await sink(inputs);
-      } else {
+      if (isEmpty(inputs)) {
```

Comment:
this definitely fixes the issue.  So the test bench apps/utils have been abstracted to the point where I don't really understand what's going on (maybe need better docs on them) - but I don't understand why do we even do anything here?  what is the "sink" in this context?  For unsafe file upload by the time the route handler is called the files have always already been parsed from the body I thought, and we would have fired or not depending on the protect mode right?  why doesn't this just always return `done` regardless of input (across all apps).

not asking for anything to change - I'm just trying to understand, b/c it is not at all obvious to me what's going on.

Reply:
yea so the UFU sink is weird. but basically the only reason it's abstracted is so we don't replicate logic across all the frameworks to indirectly test multi form parsing via reflected xss.  So if you look in `test-bench-utils/lib/routes.js` you will see
```
  inputs: ['body'],
    params: ['input'],
```

That means req.body.input will have some value that will be put into res.send to verify multi form posts can be parsed by the framework.  So getInputs will return all the params from the relevant input and in the UFU sink case just return the value of `input`

```
module.exports.upload = async function upload({ input }) {
  return input;
};
```

What broke here is getInputs used to be getInput which returned a string. It now returns an object so in the screener case it wasn't passing anything as req.body so it was an empty object which fastify apparently doesn't like you passing `res.send(undefined)`